### ISSUE TITLE
Deployment fails with: Deploy directory 'out' does not exist - when trying to deploy "serverless"

### ISSUE THREAD

**Issue by madze** [netlify/netlify-plugin-nextjs#211]
**Describe the bug**
Attempting to use the new [one-click install Next.js](https://www.netlify.com/blog/2020/12/07/announcing-one-click-install-next.js-build-plugin-on-netlify/) feature (though this actually happens even if I install the plugin directly both via netlify.toml as well as in the UI).

**I receive an error during deployment:**
>Deploy did not succeed: Deploy directory 'out' does not exist

However, I do have target set to "serverless" in the next.config.js and do not want to use ``next export`` because I have an api route that should use the netlify serverless functions.

**To Reproduce**
1. Create a next.js project on github
2. Setup CD with netlify
3. Initiate a deployment

**Expected behavior**
The site to deploy using serverless functions for SSR and the next API route.

**Versions**
- Next.js: 10.1.3

**Additional notes:**
The only solution I've been able to find when searching this error is to use ``next export`` to deploy a static site but then of course our api routes don't work.

It seems that perhaps the site just isn't being built as "serverless" for some reason...?

Here's the next.config.js:
```
module.exports = {
	target: 'serverless'
}
```
Or perhaps I need to setup my netlify account differently somehow? 

Any help would be greatly appreciated!

Thanks!

**Comment by madze** [comment]
**UPDATE:**

If create a netlify.toml and add the following to it:
```
[[plugins]]
  package = "@netlify/plugin-nextjs"

[[plugins]]
package = "netlify-plugin-cache-nextjs"

[build]
  command = "yarn build"
  publish = "out"
```

I seem to finally be getting the netlify-plugin-nextjs to run but then I get the following error:
```
Error: 'out/index.html' already exists
```

NOTE: This installs @netlify/plugin-nextjs@3.0.3

**Comment by lindsaylevine** [comment]
@madze hey! unfortunately i'll need:

1. link to your build/deploy logs (i.e. https://app.netlify.com/sites/{sitename}/deploys/{deployid})
2. a link to a simple repo that reproduces your issue 

with an error like that, i can almost guarantee it's unique to your setup so those things will help a lot! and with that error in particular, it seems like perhaps you have your own `index.html` in your publish directory already?

**Comment by madze** [comment]
@lindsaylevine - Thanks for the reply!

For some reason when the error occurred I was fixated on the "out" directory part of it (I think because of the first issue I was coming up against), but after reading your note about an existing html.file I realized that I had migrated this project over from a gatsby.js project and reused the existing public directory and of course the culprit was right there - I had just missed it when I gutted that directory of extraneous files.

I figured it had to probably be something silly that I was missing, especially because of the lack of results when I searched that error on the web.

At any rate, if anyone else happens upon a similar corner case as this, maybe this will help them as well.

Also btw, after deleting that file I was able to remove the``@netlify/plugin-nextjs`` and build commands from the netlify.toml and the one click deployment worked like a charm!

Thanks a bunch! I'll go ahead and close this ticket.
